app/models/account/states.rb

```diff
@@ -131,6 +131,10 @@ def should_be_deleted?
       suspended? || will_be_deleted?
     end
 
+    def destroyable?
+      !master? && ((buyer? && invoices.unresolved.empty?) || should_be_deleted?)
```

Comment:
In SaaS, the providers also have invoices, so they might have unresolved invoices too, that should prevent them from being deleted. I'm wondering if `buyer?` check is needed.

Reply:
This came from https://github.com/3scale/porta/blob/dd794f4fd946cc20e81a63a12650caf7d449ba2c/app/models/account/billing.rb#L42-L46

The logic is that account can always be deleted when marked for deletion but a buyer account can be deleted always as long as no unresolved invoices are present.

If we just remove the `buyer?` condition, then providers non-marked for deletion will also be deleted which would be wrong.

I don't think it makes sense to change the logic to prevent provider from being deleted for unresolved invoices. At least it was not the case until now and if we want to change this, then lets discuss in a separate PR.

Reply:
> I don't think it makes sense to change the logic to prevent provider from being deleted for unresolved invoices. At least it was not the case until now and if we want to change this, then lets discuss in a separate PR.

Hm... OK, I think the logic of this method is a bit confusing... there is no reference to the account being buyer or provider in `check_unresolved_invoices`, so I assumed that this check is for anyone, and will fail if there are pending invoices. After digging into `should_be_deleted?` I see it's not the case, but it's still confusing :upside_down_face: